apiserver/Makefile

```diff
@@ -43,16 +16,83 @@ jupyter_notebook_config.py:
 	    -e "s,@keyfile@,$(GSA_KEY_FILE),g" \
             < jupyter_notebook_config.py.in > $@
 
-$(eval $(call DOCKER_IMAGE,spark-base,install-hail-locally jupyter_notebook_config.py,SPARK_BASE_IMAGE))
-$(eval $(call DOCKER_IMAGE,hail-base,install-hail-locally jupyter_notebook_config.py,HAIL_BASE_IMAGE))
-$(eval $(call DOCKER_IMAGE,spark-master,install-hail-locally jupyter_notebook_config.py,SPARK_MASTER_IMAGE))
-$(eval $(call DOCKER_IMAGE,spark-worker,install-hail-locally jupyter_notebook_config.py,SPARK_WORKER_IMAGE))
-$(eval $(call DOCKER_IMAGE,apiserver,install-hail-locally jupyter_notebook_config.py,APISERVER_IMAGE))
-$(eval $(call DOCKER_IMAGE,hail-jupyter,install-hail-locally jupyter_notebook_config.py,HAIL_JUPYTER_IMAGE))
+build-common: install-hail-locally jupyter_notebook_config.py
```

Comment:
This should be PHONY.

Reply:
Actually, I think you should delete this.  Use these dependencies where they belong (e.g. not every docker build needs jupyter_notebook_config.py).  All docker build targets should be PHONY.  Since they'll be run every time, you don't need to explicit dependencies (unless they need to get created like for install-hail-locally), Docker caching rather than make will figure out when to rebuild.

The only reason I did that for base is that multiple images depend on base, so make base gets invoked many times.